contrib/storage-jdbc/src/main/java/org/apache/drill/exec/store/jdbc/JdbcPrel.java

```diff
@@ -130,4 +131,10 @@ public SelectionVectorMode getEncoding() {
   public boolean needsFinalColumnReordering() {
     return false;
   }
+
+  @Override
+  public Prel addImplicitRowIDCol(List<RelNode> children) {
```

Comment:
I am thinking a cleaner interface would be to have an additional method `supportsImplicitRIDCol()`  which would return FALSE by default.  You could use the Java 8 feature for default interface methods [1].   Only call `addImplicitRowIdCol()` if adding the rowid is supported.  

[1] https://dzone.com/articles/interface-default-methods-java